include/wlr/types/wlr_pointer_gestures_v1.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * This an unstable interface of wlroots. No guarantees are made regarding the
+ * future consistency of this API.
+ */
+#ifndef WLR_USE_UNSTABLE
+#error "Add -DWLR_USE_UNSTABLE to enable unstable wlroots features"
+#endif
+
+#ifndef WLR_TYPES_WLR_POINTER_GESTURES_V1_H
+#define WLR_TYPES_WLR_POINTER_GESTURES_V1_H
+
+#include <wayland-server.h>
+#include <wlr/types/wlr_pointer.h>
+#include <wlr/types/wlr_surface.h>
+
+struct wlr_pointer_gestures_v1 {
+	struct wl_global *global;
+	struct wl_list resources; // wl_resource_get_link
+	struct wl_list swipes;    // wl_resource_get_link
+	struct wl_list pinches;   // wl_resource_get_link
+
+	struct wl_listener display_destroy;
+
+	struct {
+		struct wl_signal destroy;
+	} events;
+
+	void *data;
+};
+
+struct wlr_pointer_gestures_v1 *wlr_pointer_gestures_v1_create(
+	struct wl_display *display);
+
+void wlr_pointer_gesture_swipe_notify_begin(
+	struct wlr_pointer_gestures_v1 *gestures,
+	struct wlr_surface *focus,
```

Comment:
I'd like not to make the compositor responsible for deciding which surface these events are sent to. Events should only be sent to the currently focused surface. So we should remove the surface argument and replace it with a seat argument. The focused surface is in `wlr_seat.pointer_state.focused_surface`.